src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2970,7 +2972,9 @@ private RefKind GetEffectiveParameterRefKind(ParameterSymbol parameter, RefKind
                 return Conversion.ImplicitDynamic;
             }
 
-            if (argument.Kind == BoundKind.OutVariablePendingInference || argument.Kind == BoundKind.OutDeconstructVarPendingInference)
+            if (argument.Kind == BoundKind.OutVariablePendingInference ||
+                argument.Kind == BoundKind.OutDeconstructVarPendingInference ||
+                argument.Kind == BoundKind.DiscardedExpression)
```

Comment:
>argument.Kind == BoundKind.DiscardedExpression [](start = 16, length = 46)

This doesn't look right for a discard with a type. Do we have a test with real overloading?